### PR TITLE
Improve doc for deprecated close method

### PR DESCRIPTION
**Pull request by regadas**

### CODE REVIEW

File: scio-core/src/main/scala/com/spotify/scio/ScioContext.scala [hunk 1]
```diff
@@ -653,13 +653,52 @@ class ScioContext private[scio] (
   /**
    * Runs the underlying pipeline according to the options used to create this context.
    * Running closes the context and no operation can be performed once the context is closed.
+   * This method does not respect the deprecated `--blocking` flag.
+   *
+   * This method does not respect the deprecated `--blocking` flag.
```
Comment (brianmartin) [inline scio-core/src/main/scala/com/spotify/scio/ScioContext.scala]: Did you intend both line 656 and 658?

Comment (regadas) [reply scio-core/src/main/scala/com/spotify/scio/ScioContext.scala]: oh nice catch! Thanks @brianmartin

File: scio-core/src/main/scala/com/spotify/scio/ScioContext.scala [hunk 2]
```diff
@@ -653,13 +653,43 @@ class ScioContext private[scio] (
   /**
    * Runs the underlying pipeline according to the options used to create this context.
    * Running closes the context and no operation can be performed once the context is closed.
+   *
+   * This operation is non-blocking and no operation can be performed once the context is closed.
```
Comment (nevillelyh) [inline scio-core/src/main/scala/com/spotify/scio/ScioContext.scala]: "according to the options used to create this context." is a bit redundant and implied?

Clarify `no further operation`? No further transformations?

File: scio-core/src/main/scala/com/spotify/scio/ScioContext.scala [hunk 3]
```diff
@@ -653,13 +653,43 @@ class ScioContext private[scio] (
...
+   * Runs the underlying pipeline according to the options used to create this context.
+   * Running closes the context and no operation can be performed once the context is closed.
+   *
+   * This operation is non-blocking and no operation can be performed once the context is closed.
+   *
+   * @param runner [[ScioRunner]] responsible for the execution for this context.
+   *
+   * @return the [[ScioExecutionContext]] for the underlying job execution.
+   */
   def run(runner: ScioRunner): ScioExecutionContext = runner.run(this)
```
Comment (nevillelyh) [inline scio-core/src/main/scala/com/spotify/scio/ScioContext.scala]: Add something on how this is different from the default `run()`, like explain the purpose of the runner?

File: scio-core/src/main/scala/com/spotify/scio/ScioContext.scala [hunk 4]
```diff
@@ -653,13 +653,43 @@ class ScioContext private[scio] (
...
+   * This method is deprecated and with it the `--blocking` flag.
+   *
+   * Use [[ScioContext#run]].
+   *
+   * To achieve the same behaviour when `--blocking` was enabled use:
+   *
+   * {{{
+   * val coll: SCollection = ???
+   *
+   * coll.run().waitUntilDone(Duration.Inf)
```
Comment (nevillelyh) [inline scio-core/src/main/scala/com/spotify/scio/ScioContext.scala]: Should be `sc.run()`?

Comment (regadas) [reply scio-core/src/main/scala/com/spotify/scio/ScioContext.scala]: right

File: scio-core/src/main/scala/com/spotify/scio/ScioContext.scala (hunk 3 above)
Comment (regadas) [reply scio-core/src/main/scala/com/spotify/scio/ScioContext.scala]: I plan to remove this in the stable release ... so I'll not bother with it 😄

**Pull request merged**